### PR TITLE
Add speedscale integration

### PR DESCRIPTION
**Pull request by kenahrens**
### What does this PR do?

A brief description of the change being made with this pull request.

### Motivation

What inspired you to submit this pull request?

### Review checklist

- [ ] PR has a [meaningful title](https://github.com/DataDog/integrations-core/blob/master/CONTRIBUTING.md#pull-request-title) or PR has the `no-changelog` label attached
- [ ] Feature or bugfix has tests
- [ ] Git history is clean
- [ ] If PR impacts documentation, docs team has been notified or an issue has been opened on the [documentation repo](https://github.com/DataDog/documentation/issues/new)

### Additional Notes

Anything else we should know when reviewing?

### CODE REVIEW

**Review by hithwen — CHANGES_REQUESTED** [review]
Hallo @kenahrens, thanks for your contribution. As part of adding a new integration on this repo you need to add yourself as a maintainer on `.github/CODEOWNERS` file on this same repo. This should be done as part of this same PR and it means you'll be the main responder for issues and future PRs for this integration.

File: speedscale/README.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
+# Agent Check: Speedscale
+
+## Overview
+
+This integration publishes the results from [Speedscale][1] into Datadog.
```
Comment (anawishnoff) [inline speedscale/README.md:5]: Would you be able to add a bit more detail here about the kind of data that Speedscale collects, and what types of data it's sending into Datadog?

File: speedscale/README.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+### Configuration
+
+In order to utilize the integration you need to capture the following from Datadog:
+
+* [API Key][2] - An API key is required by the Datadog Agent to submit metrics and events to Datadog
+
+A best practice is to save this into environment variables like so:
+
+```
+export DDOG_API_KEY=0
```
Comment (anawishnoff) [inline speedscale/README.md:18]: Once the user has saved their API key, where would they enter it to connect to Speedscale? It would be helpful to have some step by step instructions here.

File: speedscale/manifest.json [hunk 3]
```diff
@@ -0,0 +1,32 @@
+{
+  "display_name": "Speedscale",
+  "maintainer": "support@speedscale.com",
+  "manifest_version": "1.0.0",
+  "name": "speedscale",
+  "metric_prefix": "speedscale.",
+  "metric_to_check": "",
+  "creates_events": true,
+  "short_description": "",
```
Comment (anawishnoff) [inline speedscale/manifest.json:9]: This is a required field - it's a brief description of your offering that will be displayed on the tile.

File: speedscale/manifest.json [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  "guid": "f2e3ed64-a756-440f-9a1f-43089536aa08",
+  "support": "contrib",
+  "supported_os": [
+    "linux",
+    "mac_os",
+    "windows"
+  ],
+  "public_title": "Speedscale",
+  "categories": [
```
Comment (anawishnoff) [inline speedscale/manifest.json:18]: Categories can be found on [this page](https://docs.datadoghq.com/integrations/) in the list at the top of the page. Feel free to add any categories that are relevant to your listing.

File: speedscale/README.md (hunk 1 above)
Comment (kenahrens) [reply speedscale/README.md]: Thank you, that section was definitely too vague, I added more details.

File: speedscale/README.md (hunk 2 above)
Comment (kenahrens) [reply speedscale/README.md:18]: Thank you, I reworked this section to keep all the configuration in one place. Right now this is done through continuous integration and that is specifically called out in my latest update.

File: speedscale/manifest.json (hunk 4 above)
Comment (kenahrens) [reply speedscale/manifest.json:18]: Thank you, I was looking for that list. Perfect!

File: speedscale/README.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
+# Agent Check: Speedscale
+
+## Overview
+
+This integration publishes traffic replay results from [Speedscale][1] into Datadog. This lets you combine your observability data from Datadog with the results of a particular replay to drill-down into the root cause of poor performance. Find and troubleshoot potential performance issues before they show up in production with a combination of Speedscale and Datadog.
```
Comment (kayayarai) [inline speedscale/README.md:5]:
```suggestion
This integration publishes traffic replay results from [Speedscale][1] into Datadog. This lets you combine your observability data from Datadog with the results of a particular replay to investigate the root cause of poor performance. Find and troubleshoot potential performance issues before they show up in production with a combination of Speedscale and Datadog.
```

File: speedscale/README.md [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+
+## Setup
+
+### Configuration
+
+In order to utilize the integration you need to capture the following from Datadog:
+
+* [API Key][2] - An API key is required by the Datadog Agent to submit metrics and events to Datadog
+
+A best practice is to save this into an environment variable. Most likely you will store this environment variable in your continuous integration system, but when doing a one-off test you can access in your terminal like so:
```
Comment (kayayarai) [inline speedscale/README.md:15]:
```suggestion
1. To use this integration you need a Datadog [API Key][2] so that the Datadog Agent can submit metrics and events to Datadog.

   A best practice is to save this into an environment variable. Most likely you will store this environment variable in your continuous integration system, but when doing a one-off test you can access in your terminal like so:
```

File: speedscale/README.md [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+
+```
+export DDOG_API_KEY=0
+```
+
+Gather the report id of a specific report that you want to upload into Datadog. When working with continuous integration you should be able to get the report id associated with your commit hash. Store this report id in an environment variable:
```
Comment (kayayarai) [inline speedscale/README.md:21]:
```suggestion
2. Gather the report ID of a specific report that you want to upload into Datadog. When working with continuous integration, get the report ID associated with your commit hash. Store this report ID in an environment variable:
```

File: speedscale/README.md [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+
+```
+export SPD_REPORT_ID=0
+```
+
+Now with the specific report id and the Datadog API key, you run the `speedctl` CLI to export that traffic replay report as a Datadog event.
```
Comment (kayayarai) [inline speedscale/README.md:27]:
```suggestion
3. With the specific report ID and the Datadog API key, run the `speedctl` command to export that traffic replay report as a Datadog event.
```

File: speedscale/README.md [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+
+In order to utilize the integration you need to capture the following from Datadog:
+
+* [API Key][2] - An API key is required by the Datadog Agent to submit metrics and events to Datadog
+
+A best practice is to save this into an environment variable. Most likely you will store this environment variable in your continuous integration system, but when doing a one-off test you can access in your terminal like so:
+
+```
+export DDOG_API_KEY=0
+```
```
Comment (kayayarai) [inline speedscale/README.md:19]: indent these lines by 3 spaces to nest the codeblock inside the list item.

File: speedscale/README.md [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+
+Gather the report id of a specific report that you want to upload into Datadog. When working with continuous integration you should be able to get the report id associated with your commit hash. Store this report id in an environment variable:
+
+```
+export SPD_REPORT_ID=0
+```
```
Comment (kayayarai) [inline speedscale/README.md:25]: indent 3 spaces

File: speedscale/README.md [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+
+Now with the specific report id and the Datadog API key, you run the `speedctl` CLI to export that traffic replay report as a Datadog event.
+
+```
+speedctl export datadog report ${SPD_REPORT_ID} --apiKey ${DDOG_API_KEY}
+✔ {"status":"ok",...}
+```
```
Comment (kayayarai) [inline speedscale/README.md:32]: indent three spaces. Also, the check mark is causing an error in the file because it is non-ASCII. Is that line part of the command? or what the command returns (if this, then just delete the line).

File: speedscale/README.md [hunk 12]
```diff
@@ -0,0 +1,58 @@
...
+### Validation
+
+After you have exported the report, you should see it in the Datadog [Event Stream][2].
```
Comment (kayayarai) [inline speedscale/README.md:35]:
```suggestion
View the Datadog [Event Stream][2] to see your exported report.
```

File: speedscale/README.md [hunk 13]
```diff
@@ -0,0 +1,58 @@
...
+
+Speedscale does not include any metrics.
+
+### Service Checks
+
+Speedscale does not include any service checks.
+
+### Events
+
+The Speedscale integration sends events to your [Datadog Event Stream][3] when a traffic replay is complete to understand the impact this has on your metrics.
```
Comment (kayayarai) [inline speedscale/README.md:49]:
```suggestion
The Speedscale integration sends events to your [Datadog Event Stream][3] when a traffic replay is complete to help you understand the impact this has on your metrics.
```
Not sure I understand what "the impact this has on your metrics" means. What's the "this"? Maybe rephrase to be specific?

File: speedscale/manifest.json [hunk 14]
```diff
@@ -0,0 +1,35 @@
+{
+  "display_name": "Speedscale",
+  "maintainer": "support@speedscale.com",
+  "manifest_version": "1.0.0",
+  "name": "speedscale",
+  "metric_prefix": "speedscale.",
+  "metric_to_check": "",
+  "creates_events": true,
+  "short_description": "This integration publishes traffic replay results from Speedscale into Datadog.",
```
Comment (kayayarai) [inline speedscale/manifest.json:9]:
```suggestion
  "short_description": "Publish traffic replay results from Speedscale into Datadog.",
```

**Review by kayayarai — CHANGES_REQUESTED** [review]
Thanks for this contribution! I made some general wording edits, and turned the config process into numbered steps. And I left some questions for you to address.

File: speedscale/README.md [hunk 15]
```diff
@@ -0,0 +1,56 @@
+# Agent Check: Speedscale
+
+## Overview
+
+This integration publishes traffic replay results from [Speedscale][1] into Datadog. This lets you combine your observability data from Datadog with the results of a particular replay to investigate the root cause of poor performance. Find and troubleshoot potential performance issues before they show up in production with a combination of Speedscale and Datadog.
```
Comment (anawishnoff) [inline speedscale/README.md:5]:
```suggestion
This integration publishes traffic replay results from [Speedscale][1] into Datadog. This lets you combine your observability data from Datadog with the results of a particular Speedscale replay to investigate the root cause of poor performance. Find and troubleshoot potential performance issues before they show up in production with the Speedscale Datadog integration.
```

File: speedscale/README.md [hunk 16]
```diff
@@ -0,0 +1,56 @@
...
+
+## Setup
+
+### Configuration
+
+1. To use this integration you need a Datadog [API Key][2] so that the Datadog Agent can submit metrics and events to Datadog.
```
Comment (anawishnoff) [inline speedscale/README.md:11]:
```suggestion
1. To use this integration you need a Datadog [API Key][2] so that metrics and events can be submitted into Datadog.
```
Correct me if I'm wrong, but I thought that this integration did not use the Datadog agent?

File: speedscale/README.md [hunk 17]
```diff
@@ -0,0 +1,56 @@
...
+
+   A best practice is to save this into an environment variable. Most likely you will store this environment variable in your continuous integration system, but when doing a one-off test you can access in your terminal like so:
```
Comment (anawishnoff) [inline speedscale/README.md:13]:
```suggestion
A best practice is to save this value into an environment variable. Most likely you will store this environment variable in your continuous integration system, but when doing a one-off test you can access it in your terminal like so:
```

File: speedscale/README.md [hunk 18]
```diff
@@ -0,0 +1,56 @@
...
+
+```
```
Comment (anawishnoff) [inline speedscale/README.md:15]:
```suggestion
    ```
```

File: speedscale/README.md [hunk 19]
```diff
@@ -0,0 +1,56 @@
...
+export DDOG_API_KEY=0
+```
+
+2. Gather the report ID of a specific report that you want to upload into Datadog. When working with continuous integration, get the report ID associated with your commit hash. Store this report ID in an environment variable:
```
Comment (anawishnoff) [inline speedscale/README.md:19]:
```suggestion
2. Gather the report ID of a specific report that you'd like to upload to Datadog. When working with continuous integration, get the report ID associated with your commit hash. Store this report ID in an environment variable:
```

**Comment by anawishnoff** [comment]
Hi @kenahrens, just checking in on the progress of this PR. Let us know if you have any questions about the suggested changes. Thanks!

**Comment by kenahrens** [comment]
> Hi @kenahrens, just checking in on the progress of this PR. Let us know if you have any questions about the suggested changes. Thanks!

Hi @anawishnoff - I have no questions about the changes, I will pull them in and also add a dashboard as well. It just slipped a little because of timing on our end. Thanks!

**Comment by kenahrens** [comment]
Hi @anawishnoff - my apologies again for the delay, I have merged all the suggested changes and spacing / improvements and added a dashboard as well. I'm not totally familiar with the process, I requested reviews again from your team.

Here is an example of the dashboard I was testing:
Screen Shot 2022-04-27 at 6 37 38 AM

File: speedscale/manifest.json [hunk 20]
```diff
@@ -0,0 +1,35 @@
...
+    "automation",
+    "containers",
+    "orchestration",
+    "testing"
+  ],
+  "type": "crawler",
+  "is_public": false,
+  "integration_id": "speedscale",
+  "assets": {
+    "dashboards": {},
```
Comment (anawishnoff) [inline speedscale/manifest.json:28]: You'll need to add a reference here to the dashboard you created

File: speedscale/README.md [hunk 21]
```diff
@@ -0,0 +1,56 @@
+# Agent Check: Speedscale
```
Comment (anawishnoff) [inline speedscale/README.md:1]: I'd recommend removing this line since the integration is not using the Agent

Comment (kenahrens) [reply speedscale/README.md]: updated

File: speedscale/assets/dashboards/SpeedscaleOverview.json [hunk 22]
```diff
@@ -0,0 +1 @@
...ut when doing a one-off test you can access it in your terminal like so:\n\n   ```\n   export DDOG_API_KEY=0\n   ```\n\n2. Gather the report ID of a specific report that you'd like to upload to Datadog. When working with continuous integration, get the report ID associated with your commit hash. Store this report ID in an environment variable:\n\n   ```\n   export SPD_REPORT_ID=0\n   ```\n\n3. With the specific report ID and the Datadog API key, run the `speedctl` command to export that traffic replay report as a Datadog event.\n\n   ```\n   speedctl export datadog report ${SPD_REPORT_ID} --apiKey ${DDOG_API_KEY}\n   {\"status\":\"ok\",...}\n   ```","background_color":"white","font_size":"14","text_align":"left","vertical_align":"top","show_tick":false,"tick_pos":"50%","tick_edge":"left","has_padding":true},"layout":{"x":8,"y":0,"width":4,"height":6}},{"id":2427201979903764,"definition":{"title":"","title_size":"16","title_align":"left","type":"event_timeline","query":"source:speedscale","tags_execution":"and"},"layout":{"x":0,"y":4,"width":8,"height":2}}],"template_variables":[],"layout_type":"ordered","is_read_only":false,"notify_list":[],"reflow_type":"fixed","id":"pb4-v54-h4h"}
```
[the remark this answers is not in the document]
Comment (kenahrens) [reply speedscale/assets/dashboards/SpeedscaleOverview.json]: I put in your suggestions and removed the setup instructions

**Review by anawishnoff — APPROVED** [review]
Looks good from product :)

File: speedscale/manifest.json [hunk 23]
```diff
@@ -0,0 +1,37 @@
...
+  ],
+  "public_title": "Speedscale",
+  "categories": [
+    "automation",
+    "containers",
+    "orchestration",
+    "testing"
+  ],
+  "type": "crawler",
+  "is_public": false,
```
Comment (yzhan289) [inline speedscale/manifest.json:25]: `is_public` is currently set to `false`, so this tile won't show up on Datadog docs. If you want this to appear in https://docs.datadoghq.com/integrations/, you should set this to `true` instead.

Comment (kenahrens) [reply speedscale/manifest.json]: Thank you for the catch, I just updated.

**Review by yzhan289 — APPROVED** [review]
PR looks good to me. We are currently requesting a logo internally for Speedscale, so once that's ready we can merge this and deploy it to production.
